android/app/src/main/java/updated/mysterium/vpn/core/NodeRepository.kt

```diff
@@ -156,13 +156,15 @@ class NodeRepository(var deferredNode: DeferredNode) {
                 val order = deferredNode.await().createPaymentGatewayOrder(req).decodeToString()
                 Log.d(TAG, "createPaymentOrder response: $order")
                 Order.fromJSON(order) ?: error("Could not parse JSON: $order")
-            } catch (e: Exception) {
+            } catch (exception: Exception) {
                 if (isBalanceLimitExceeded()) {
-                    throw TopupPreconditionFailedException(
-                        e.message ?: "You can only top-up if you have less than 5 MYST in balance"
+                    throw TopupBalanceLimitException(
+                        exception.message ?: "You can only top-up if you have less than 5 MYST in balance"
                     )
+                } else if (exception.message == "Cannot provide more balance at this time") {
```

Comment:
It's a bad practise to check error message. But if we already do - let's move messages to constants. we can do it exactly in companion object of NodeRepository or create some object like `RowErrorMessage` and move error messages there.

Reply:
Done